lib/stats/stats-cluster.c

```diff
@@ -261,22 +289,29 @@ stats_counter_group_init_equals(const StatsCounterGroupInit *self, const StatsCo
 gboolean
 stats_cluster_key_equal(const StatsClusterKey *key1, const StatsClusterKey *key2)
 {
-  return key1->legacy.component == key2->legacy.component
-         && strcmp(key1->id, key2->id) == 0
-         && strcmp(key1->legacy.instance, key2->legacy.instance) == 0
-         && stats_counter_group_init_equals(&key1->counter_group_init, &key2->counter_group_init)
-         && stats_cluster_key_labels_equal(key1->labels, key1->labels_len, key2->labels, key2->labels_len);
+  if (stats_cluster_key_is_legacy(key1) != stats_cluster_key_is_legacy(key2))
+    return FALSE;
+
+  if (stats_cluster_key_is_legacy(key1) && stats_cluster_key_is_legacy(key2))
```

Comment:
[optional]

At this point we know, that the two `is_legacy()` are the same, so it is enough to check one of them.